Sources/OpenAPIRuntime/Errors/ClientError.swift

```diff
@@ -64,6 +64,10 @@ public struct ClientError: Error {
     /// Will be nil if the error resulted before the response was received.
     public var responseBody: HTTPBody?
 
+    /// A user-facing description of what caused the underlying error
+    /// to be thrown.
+    public var causeDescription: String
```

Comment:
Why use a `causeDescription` over wrapping the actual underlying cause in another error which represents the cause description? 

For example, with the middleware case, I can imagine something like:

```swift
struct MiddlewareError: Error {
  var middlewareType: String
  var underlyingError: any Error

  var description: String { 
    "Middleware of type '(middlewareType)' threw ..."
  }
}
```

Reply:
We'd have to introduce several new public types then, one for Transport, another for Middleware, and so on. But the use case doesn't require it. Folks just want to get the underlying root error to handle programmatically, and they want to be able to log the cause (but not handle it programmatically).

For code internal to a package I'd probably do what you suggest, but here since we're crossing an API stable boundary, we don't want to expand the surface unless it's really justified. (For example, we don't want folks programmatically handling the `middlewareType` for the same reason we made RuntimeError internal and not public - as it's inherently unstable, so it should only be consumed in logs).